app/controllers/medications_controller.rb

```diff
@@ -0,0 +1,27 @@
+class MedicationsController < ApplicationController
+  def new; end
+
+  def search
+    results = SEARCH_RESULTS.med_search_results(params[:medication_name])
+    return @med_hash = results unless results.nil?
+    flash[:warning] = "Sorry, your search did not return any results. Please try another search."
+    redirect_to '/medications/new'
+  end
```

Comment:
we could consider refactoring this action into a `MedicationsSearchController#index action` to follow ReST

Reply:
well technically `SEARCH_RESULTS` is just an instance of `SearchResultsFacade` and can be used anywhere we need it to be used :) It's just a class, and Rails magic makes it accessible anywhere.
but yeah, this could be a different controller